pyfstat/make_sfts.py

```diff
@@ -129,10 +117,78 @@ def basic_setup(self):
             )
             for ind, dets in enumerate(IFOs)
         ]
+        self.IFOs = ",".join(['"{}"'.format(d) for d in IFOs])
+
+    def _get_setup_from_noiseSFTs(self):
+        noise_multi_sft_catalog = lalpulsar.GetMultiSFTCatalogView(
+            lalpulsar.SFTdataFind(self.noiseSFTs, self.SFTconstraint)
+        )
+
+        # Information to be extracted from the SFTs themselves
+        IFOs = []
+        tstart = []
+        tend = []
+        Tsft = []
+        self.sftfilenames = []  # This refers to the MFD output!
+
+        # SWIG-LAL does not import this functionality
+        gps_to_float = lambda x: x.gpsSeconds
+
+        for ifo_catalog in noise_multi_sft_catalog.data:
+            ifo_name = lalpulsar.ListIFOsInCatalog(ifo_catalog).data[0]
+
+            time_stamps = lalpulsar.TimestampsFromSFTCatalog(ifo_catalog)
+            this_start_time = gps_to_float(time_stamps.data[0])
+            this_end_time = gps_to_float(time_stamps.data[-1])
+            this_Tsft = int(round(1.0 / ifo_catalog.data[0].header.deltaF))
+
+            self.sftfilenames.append(
+                lalpulsar.OfficialSFTFilename(
+                    ifo_name[0],
+                    ifo_name[1],
+                    ifo_catalog.length,
+                    this_Tsft,
+                    this_start_time,
+                    this_end_time - this_start_time + this_Tsft,
+                    self.label,
+                )
+            )
+
+            IFOs.append(ifo_name)
+            tstart.append(this_start_time)
+            tend.append(this_end_time)
+            Tsft.append(this_Tsft)
+
+        # Get the "overall" values of the search
+        self.tstart = min(tstart)
+        self.tend = max(tend)
+        self.duration = self.tend - self.tstart + this_Tsft
+        self.IFOs = ",".join(['"{}"'.format(d) for d in IFOs])
+        Tsft = np.unique(Tsft)
+        if len(Tsft) != 1:
+            raise ValueError("SFTs contain different basetimes: {}".format(Tsft))
+        else:
+            self.Tsft = Tsft[0]
+
+    def basic_setup(self):
+        os.makedirs(self.outdir, exist_ok=True)
+        self.config_file_name = os.path.join(self.outdir, self.label + ".cff")
+        self.theta = np.array([self.phi, self.F0, self.F1, self.F2])
+
+        if self.noiseSFTs is not None:
+            logging.warning(
```

Comment:
@dbkeitel opinion on how stronger this warning should be?

Would you set "0-like" default values for everything?

Reply:
I think this only has to be a warning if the other options are explicitly set - so yes, making them `None` by default would make sense.

However I'm wondering if it would also be good to be able to use noiseSFTs but still override tstart, duration and detectors with the existing keywords, by having PyFstat assemble the constraints itself.